### PR TITLE
Fix plugin install exit code when install fails

### LINKED ISSUE #2470: CLI plugin install command exits with zero on errors
From #2469, separate bug:

```
/kontena/test # kontena plugin install aws && echo yay it worked
    *   Installing plugin aws .. /Called 'load' without the :safe option -- defaulting to safe mode.
You can avoid this warning in the future by setting the SafeYAML::OPTIONS[:default_mode] option (to :safe or :unsafe).
/Gem::InstallError : invalid gem: package is corrupt, exception while verifying: YAML data doesn't evaluate to gem specification (Gem::Exception) in /root/.kontena/gems/2.3.3/cache/kontena-cli-1.3.0.gem
 [fail] Installing plugin aws     
yay it worked
```

### PR DESCRIPTION
**Pull request by kke**
Fixes #2470

### CODE REVIEW

File: cli/lib/kontena/cli/plugins/install_command.rb [hunk 1]
```diff
@@ -34,11 +35,14 @@ def execute
...
             $stderr.puts pastel.red("#{ex.class.name} : #{ex.message}")
             logger.error(ex)
-            spin.fail!
+            spin.fail
+            ex
           end
         end
       end
 
+      exit_with_error "Install failed: #{installed.message}" if installed.respond_to?(:message)
```
Comment (SpComb) [inline cli/lib/kontena/cli/plugins/install_command.rb]: Why not just let the exception raise itself instead?

Comment (kke) [inline cli/lib/kontena/cli/plugins/install_command.rb]: It does that now

**Comment by SpComb** [comment]
Unable to review with the #2462 changes merged in.

I think the best way to do chained PRs on github would be to set the #2462 branch as the base for this PR, then github will show the diff against that PR. Once the base PR is merged, fix the base branch for this PR back to master.

**Comment by kke** [comment]
Cleaned out the other PR

**Comment by kke** [comment]
The rubygems commands/functions/whatever do write to STDOUT during spinning and mess out the output, but that's another issue for the future.

**Pull request merged**